core/interop/tests/module.rs

```diff
@@ -0,0 +1,112 @@
+//! Test for the class proc-macro.
+#![allow(unused_crate_dependencies)]
+
+use boa_engine::{js_string, Context, JsString, Module, Source};
+use boa_macros::{boa_class, boa_module, Finalize, JsData, Trace};
+use std::rc::Rc;
+
+#[derive(Clone, Trace, Finalize, JsData)]
+enum AnimalType {
+    Cat,
+    Dog,
+    Other,
+}
+
+#[derive(Clone, Trace, Finalize, JsData)]
+struct Animal {
+    ty: AnimalType,
+    age: i32,
+}
+
+#[boa_class]
+#[boa(rename = "camelCase")]
+impl Animal {
+    #[boa(constructor)]
+    #[allow(clippy::needless_pass_by_value)]
+    fn new(name: String, age: i32) -> Self {
+        let ty = match name.as_str() {
+            "cat" => AnimalType::Cat,
+            "dog" => AnimalType::Dog,
+            _ => AnimalType::Other,
+        };
+
+        Self { ty, age }
+    }
+
+    #[boa(getter)]
+    fn age(&self) -> i32 {
+        self.age
+    }
+
+    fn speak(#[boa(error = "`this` was not an animal")] &self) -> JsString {
+        match self.ty {
+            AnimalType::Cat => js_string!("meow"),
+            AnimalType::Dog => js_string!("woof"),
+            AnimalType::Other => js_string!(r"¯\_(ツ)_/¯"),
+        }
+    }
+}
+
+#[boa_module]
+#[boa(rename = "camelCase")]
+#[boa(rename_class = "PascalCase")]
```

Comment:
Is it really necessary to always specify the naming schema? IMO it makes much more sense to automatically rename to the "correct" schema, then just provide a generic `#[boa(js_name = "newName")]` if the user wants to override the default name. Do you think it is too surprising to automatically change the name of exports and class members?

Reply:
`#[boa(name = "...")]` is available on all members, so this could definitely be done. I was thinking about just making `camelCase` the default for const and functions, and `PascalCase` the default for classes. WDYT?

Reply:
Yes, that sounds nice!